commit 92b1e7cc6766023050aaaae41390bb5cfcc07ec5
Author: Max Lin <max7442@gmail.com>
Date:   Thu Nov 3 11:48:32 2011 +0800

    Used the gps_open() return value as condition instead of checking m_gpsdata
    
    Checking the return value of gps_open() as condition instead of checking
    m_gpsdata for GPSD_API_MAJOR_VERSION >=5, there should be prevent crash
    occurred when no gpsd running.
    
    BUG: 277036
    REVIEW: 103035

diff --git a/plasma/generic/dataengines/geolocation/location_gps.cpp b/plasma/generic/dataengines/geolocation/location_gps.cpp
index 709321b..551aec2 100644
--- a/plasma/generic/dataengines/geolocation/location_gps.cpp
+++ b/plasma/generic/dataengines/geolocation/location_gps.cpp
@@ -83,11 +83,11 @@ Gps::Gps(QObject* parent, const QVariantList& args)
 {
 #if GPSD_API_MAJOR_VERSION >= 5
     m_gpsdata = new gps_data_t;
-    gps_open("localhost", DEFAULT_GPSD_PORT, m_gpsdata);
+    if (gps_open("localhost", DEFAULT_GPSD_PORT, m_gpsdata) != -1) {
 #else
     gps_data_t* m_gpsdata = gps_open("localhost", DEFAULT_GPSD_PORT);
-#endif
     if (m_gpsdata) {
+#endif
         kDebug() << "gpsd found.";
         m_gpsd = new Gpsd(m_gpsdata);
         connect(m_gpsd, SIGNAL(dataReady(Plasma::DataEngine::Data)),
